### PR TITLE
Misc code cleanups

### PR DESCRIPTION
**Pull request by mp15**
Replace != with < in for loops in errmod.
Replace 0's with NULL in bam_plcmd.c
Replace exit(1)'s with exit(EXIT_FAILURE)'s in bam_plcmd.c.

### CODE REVIEW

File: bam_plcmd.c [hunk 1]
```diff
@@ -276,13 +276,13 @@ static void group_smpl(mplp_pileup_t *m, bam_sample_t *sm, kstring_t *buf,
             const bam_pileup1_t *p = plp[i] + j;
             uint8_t *q;
             int id = -1;
-            q = ignore_rg? 0 : bam_aux_get(p->b, "RG");
+            q = ignore_rg ? NULL : bam_aux_get(p->b, "RG");
```
Comment (jmarshall) [inline bam_plcmd.c]: Okay re `NULL`, but the space before `?` is a stylistic choice.  In samtools the existing style is 80% without; in bam_plcmd.c it's exclusively without.  So please leave the existing style here. &lt;/pedantic-picard-hat&gt;

File: bam_plcmd.c [hunk 2]
```diff
@@ -276,13 +276,13 @@ static void group_smpl(mplp_pileup_t *m, bam_sample_t *sm, kstring_t *buf,
...
             if (q) id = bam_smpl_rg2smid(sm, fn[i], (char*)q+1, buf);
             if (id < 0) id = bam_smpl_rg2smid(sm, fn[i], 0, buf);
             if (id < 0 || id >= m->n) {
                 assert(q); // otherwise a bug
                 fprintf(stderr, "[%s] Read group %s used in file %s but absent from the header or an alignment missing read group.\n", __func__, (char*)q+1, fn[i]);
-                exit(1);
+                exit(EXIT_FAILURE);
```
Comment (jmarshall) [inline bam_plcmd.c]: `EXIT_FAILURE` is supplied by stdlib.h.  Presumably bam_plcmd.c gets it by accident from some header; but please add also a `#include` to bam_plcmd.c to make sure.

File: bam_plcmd.c (hunk 1 above)
Comment (mp15) [inline bam_plcmd.c]: Fixed.

File: bam_plcmd.c (hunk 2 above)
Comment (mp15) [inline bam_plcmd.c]: Done

**Comment by jmarshall** [comment]
Thanks, though let's not get distracted from the significant stuff at this stage.  Cherry-picked as b8ff342fd5846c5df6129d67b154aa8c1a157d8f.

**Pull request closed without merging**